apps/wallet/src/ui/app/pages/qredo-connect/components/SelectQredoAccountsSummaryCard.tsx

```diff
@@ -0,0 +1,137 @@
+                    return acc;
+                },
+                {}
+            ),
+        [selectedAccounts]
+    );
+    const { data, isLoading, error } = useFetchQredoAccounts(
+        qredoID,
+        fetchAccountsEnabled
+    );
+    const [searchTerm, setSearchTerm] = useState('');
+    return (
+        <SummaryCard
+            header="Select accounts"
+            body={
+                <Loading loading={isLoading}>
+                    {error ? (
+                        <Alert>
+                            Failed to fetch accounts. Please try again later.
+                        </Alert>
+                    ) : data?.length ? (
+                        <>
+                            <div className="flex items-center bg-white pt-1 sticky -top-4">
+                                <input
+                                    className="flex-1 p-3 pr-7.5 bg-white border border-solid border-gray-45 rounded-lg text-steel-dark placeholder:text-steel"
+                                    onChange={(e) =>
+                                        setSearchTerm(e.target.value)
+                                    }
+                                    value={searchTerm}
+                                    placeholder="Search"
+                                />
+                                <Search16 className="absolute w-4.5 h-4.5 right-3 pointer-events-none text-steel" />
+                            </div>
+                            <div className="divide-x-0 divide-y divide-gray-40 divide-solid">
+                                {data
+                                    .filter((aWallet) =>
+                                        matchesSearchTerm(aWallet, searchTerm)
+                                    )
+                                    .map((wallet) => (
+                                        <QredoAccountItem
+                                            key={wallet.walletID}
+                                            {...wallet}
+                                            selected={
+                                                !!selectedAccountsIndex[
+                                                    wallet.walletID
+                                                ]
+                                            }
+                                            onClick={() => {
+                                                const newSelected =
+                                                    !selectedAccountsIndex[
+                                                        wallet.walletID
+                                                    ];
+                                                onChange(
+                                                    data.filter(
+                                                        ({ walletID }) =>
+                                                            walletID ===
+                                                            wallet.walletID
```

Comment:
nit: the nesting makes this difficult to read through, I wonder if we could component-ize this a little better so we don't have to stuff everything inline under the nested ternary logic